addons/website/models/website.py

```diff
@@ -1811,20 +1818,28 @@ def _trigram_enumerate_words(self, search_details, search, limit):
                         search=unaccent(sql.Placeholder('search')),
                         field=unaccent(sql.SQL('value'))
                     )
+                    where_clause_domain = """WHERE lang = {lang}
+                                           AND name = ANY({names})
+                                           AND type = 'model'
+                                           AND value IS NOT NULL
+                                           """
+                    # This method uses hardcoded sql query, it needs filter regarding performance.
+                    # In this case for the is_published field as customer don't have access to
+                    # unpublished objects.
+                    if 'is_published' in model._fields and not self.env.user.has_group('base.group_user'):
+                        where_clause_domain += f"AND res_id in (select id from {model._table} where is_published) "
+                    where_clause = sql.SQL(where_clause_domain).format(lang=sql.Placeholder('lang'),
+                                                                      names=sql.Placeholder('names'))
```

Comment:
```suggestion
                    where_clause = sql.SQL(where_clause).format(lang=sql.Placeholder('lang'),
                                                                      names=sql.Placeholder('names'))
```
And rename `where_clause_domain` by `where_clause` in the lines above, why using 2 different variables?
Also, please try to format it so we don't waste space and start line at 80+ char
```suggestion
                    where_clause = sql.SQL(where_clause).format(
                        lang=sql.Placeholder('lang'),
                        names=sql.Placeholder('names'),
                    )
```